terraform/vpc.tf

```diff
@@ -0,0 +1,117 @@
+data "aws_availability_zones" "iac-azs" {
+    state = "available"
+}
+
+resource "aws_vpc" "da-wordpress-vpc" {
+  cidr_block = "${var.cidr_vpc}"
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+  tags = {
+    Name = "da-wordpress-vpc"
+  }
+}
+
+resource "aws_eip" "da-wordpress-eip" {
+  vpc=true
+}
+
+resource "aws_internet_gateway" "da-wordpress-igw" {
+  vpc_id = "${aws_vpc.da-wordpress-vpc.id}"
+  tags = {
+    Name ="wp-igw"
+  }
+}
+
+
+resource "aws_subnet" "private-wp-a" {
+  vpc_id                  = "${aws_vpc.da-wordpress-vpc.id}"
+  cidr_block              = "${var.private_subnet-wp-a}"
+  availability_zone       = "${data.aws_availability_zones.iac-azs.names[0]}"
+
+  tags = {
+    Name = "private-subnet-a"
+  }
+}
+
+resource "aws_subnet" "private-wp-b" {
+  vpc_id                  = "${aws_vpc.da-wordpress-vpc.id}"
+  cidr_block              = "${var.private_subnet-wp-b}"
+  availability_zone       = "${data.aws_availability_zones.iac-azs.names[1]}"
+
+  tags = {
+    Name = "private-subnet-b"
+  }
+}
+
+resource "aws_subnet" "public-wp-a" {
+  vpc_id                  = "${aws_vpc.da-wordpress-vpc.id}"
+  cidr_block              = "${var.public_subnet-wp-a}"
+  availability_zone       = "${data.aws_availability_zones.iac-azs.names[2]}"
+
+  tags = {
+    Name = "public-subnet-a"
+  }
+}
+
+resource "aws_subnet" "public-wp-b" {
+  vpc_id                  = "${aws_vpc.da-wordpress-vpc.id}"
+  cidr_block              = "${var.public_subnet-wp-b}"
+  availability_zone       = "${data.aws_availability_zones.iac-azs.names[0]}"
+
+  tags = {
+    Name = "public-subnet-b"
+  }
+}
+
```

Comment:
Here we can optimize the syntax as well:

```
####### define the subnets #######
resource "aws_subnet" "public-subnet-iac" {
  count = "${length(var.subnet_cidrs_public)}"

  vpc_id = "${aws_vpc.devopsacademy-vpc-iac.id}"
  cidr_block = "${var.subnet_cidrs_public[count.index]}"
  availability_zone = "${data.aws_availability_zones.available.names[count.index]}"

  tags = {
    Name = "public-${data.aws_availability_zones.available.names[count.index]}"
  }
}
resource "aws_subnet" "private-subnet-iac" {
  count = "${length(var.subnet_cidrs_private)}"

  vpc_id = "${aws_vpc.devopsacademy-vpc-iac.id}"
  cidr_block = "${var.subnet_cidrs_private[count.index]}"
  availability_zone = "${data.aws_availability_zones.available.names[count.index]}"

  tags = {
    Name = "private-${data.aws_availability_zones.available.names[count.index]}"
  }
}
```

Reply:
Hi Adriana, at this stage I am focusing on my task, which is to meet the requirements for the VPC (main file) and making sure that works. I am sure there are plenty of changes that can be made later on.